ci: Enable experimental kernel stuff in most CI tasks via `dev-mode`

maflcko opened the pull request:
Most of the CI tasks have a long list of stuff that they enable. This makes it hard to see what each CI task is actually running.

Also, most of the CI tasks should probably mimic the `dev-mode` CMake preset and run on as much stuff as possible. Usually, changing the `dev-mode` comes with changing those CI tasks as well in the same commit, which is verbose.

Fix both issues, by basing most CI tasks on the `dev-mode`. In the future, this makes it easier to change the `dev-mode` in a single place. If CI tasks explicitly disable something, it will be listed explicitly in them.

As a side-effect this will enable the kernel stuff for some CI task that did not have it enabled, which seems desirable.

Commit:
ci: [refactor] Use --preset=dev-mode in mac_native task

Also shorten the name, because it is usually truncated anyway in the web
view.

USDT remains disabled explicitly.

Commit:
ci: Enable experimental kernel stuff in MSan task

Base the task on --preset=dev-mode to ensure maximal coverage and add
the following:

   bitcoin-chainstate (experimental) ... ON
   libbitcoinkernel (experimental) ..... ON
   kernel-test (experimental) .......... ON

The GUI remains disabled explicitly.

Commit:
ci: [refactor] Base nowallet task on --preset=dev-mode

This makes it clearer what pieces are disabled over the full dev-mode.

The wallet remains explicitly disabled.

Commit:
ci: Enable experimental kernel stuff in TSan task

Base the task on --preset=dev-mode to ensure maximal coverage and add
the following:

   bitcoin-chainstate (experimental) ... ON
   libbitcoinkernel (experimental) ..... ON
   kernel-test (experimental) .......... ON

The GUI remains disabled explicitly.

Commit:
ci: Enable experimental kernel stuff in valgrind task

Base the task on --preset=dev-mode to ensure maximal coverage and add
the following:

   bitcoin-chainstate (experimental) ... ON
   libbitcoinkernel (experimental) ..... ON
   kernel-test (experimental) .......... ON

The GUI and USDT remain disabled explicitly.

Commit:
ci: Enable experimental kernel stuff in s390x task

Base the task on --preset=dev-mode to ensure maximal coverage and add
the following:

   bitcoin-chainstate (experimental) ... ON
   libbitcoinkernel (experimental) ..... ON
   kernel-test (experimental) .......... ON

maflcko commented:
Hmm. Looks like this uncovered some errors. I'll circle back next week.

TheCharlatan commented:
Thanks for going through this!

My guess is the failure in the previous releases job is gcc11's limited implementation of std::ranges. Not sure what to do about that to be honest. Maybe we can add a macro guard for those test cases?

The test case producing the ubsan error should just be deleted in my opinion. It does not add anything in terms of coverage. How about:
```diff
diff --git a/src/test/kernel/test_kernel.cpp b/src/test/kernel/test_kernel.cpp
index d9875ee16e..b170ea182d 100644
--- a/src/test/kernel/test_kernel.cpp
+++ b/src/test/kernel/test_kernel.cpp
@@ -583,0 +584 @@ BOOST_AUTO_TEST_CASE(btck_block)
+    BOOST_CHECK_THROW(Block{hex_string_to_byte_vec("012300")}, std::runtime_error);
@@ -716,11 +716,0 @@ void chainman_mainnet_validation_test(TestDirectory& test_directory)
-    {
-        // Process an invalid block
-        auto raw_block = hex_string_to_byte_vec("012300");
-        BOOST_CHECK_THROW(Block{raw_block}, std::runtime_error);
-    }
-    {
-        // Process an empty block
-        auto raw_block = hex_string_to_byte_vec("");
-        BOOST_CHECK_THROW(Block{raw_block}, std::runtime_error);
-    }
-
```

On a related note, there is some ongoing work at improving the test binary. Initially it was kept isolated from the rest of the test framework to demonstrate its stand-alone capabilities. I don't think that is beneficial anymore - we should just use our existing test utilities. There is also an attempt at standardizing the test vectors such that other bindings implementations can re-use them.

TheCharlatan commented:
> I was thinking about permitting nullptr when size=0, or even just permitting nullptr for any size, when the data pointer is followed by a size?

Afaict our de-serialization can handle this case (nullptr when size=0) correctly, but fails when passed a nullptr with non-zero size. Maybe a null check in `btck_block_create(...)` would be better? What do you think?

Commit:
ci: Enable experimental kernel stuff in i686 task

Base the task on --preset=dev-mode to ensure maximal coverage and add
the following:

   bitcoin-chainstate (experimental) ... ON
   libbitcoinkernel (experimental) ..... ON
   kernel-test (experimental) .......... ON

IPC remains explicitly disabled.

Commit:
ci: Enable experimental kernel stuff in Alpine task

Base the task on --preset=dev-mode to ensure maximal coverage and add
the following:

   bitcoin-chainstate (experimental) ... ON
   libbitcoinkernel (experimental) ..... ON
   kernel-test (experimental) .......... ON

Commit:
ci: Enable experimental kernel stuff in mac-cross tasks

Base the task on --preset=dev-mode to ensure maximal coverage and add
the following:

   bitcoin-chainstate (experimental) ... ON
   libbitcoinkernel (experimental) ..... ON
   kernel-test (experimental) .......... ON

USDT remains explicitly disabled.

Commit:
ci: Enable experimental kernel stuff in armhf task

Base the task on --preset=dev-mode to ensure maximal coverage and add
the following:

   bitcoin-chainstate (experimental) ... ON
   libbitcoinkernel (experimental) ..... ON
   kernel-test (experimental) .......... ON

Commit:
ci: Enable bitcoin-chainstate and test_bitcoin-qt in win64 task

Base the task on --preset=dev-mode to ensure maximal coverage and add
the following:

   bitcoin-chainstate (experimental) ... ON
   test_bitcoin-qt ..................... ON

IPC and USDT remain explicitly disabled.

sedited reviewed (approved):
Nice, ACK fae83611b8ef358ea7aca7070fd7e82dc06f9755

janb84 reviewed (commented):
ACK fae83611b8ef358ea7aca7070fd7e82dc06f9755

The PR changes CI Tasks to use the dev-mode by default and disable options that are not applicable. This to enable kernel options from libbitcoinkernel

I like the approach of "enabling everything" and then turn of options explicitly , in this way the CI will build all the options without a lot of maintenance.

I have compared the resulting config with the old configuration , looks OK 2 non blocking questions:

ARM32 diff is only kernel stuffs, OK
<details><summary>ARM32</summary>

OLD:

```sh
Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... ON
  bitcoin-qt (GUI) .................... ON
  bitcoin-gui (GUI, multiprocess) ..... ON
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... ON
  bitcoin-chainstate (experimental) ... OFF
  libbitcoinkernel (experimental) ..... OFF
  kernel-test (experimental) .......... OFF
Optional features:
  wallet support ...................... ON
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. ON
  USDT tracing ........................ ON
  QR code (GUI) ....................... ON
  DBus (GUI) .......................... ON
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... ON
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
 ```
 
  NEW:
  
  ```sh
  Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... ON
  bitcoin-qt (GUI) .................... ON
  bitcoin-gui (GUI, multiprocess) ..... ON
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... ON
  bitcoin-chainstate (experimental) ... ON
  libbitcoinkernel (experimental) ..... ON
  kernel-test (experimental) .......... ON
Optional features:
  wallet support ...................... ON
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. ON
  USDT tracing ........................ ON
  QR code (GUI) ....................... ON
  DBus (GUI) .......................... ON
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... ON
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
  ```
  
  </details>
  
  i686, no IPC diff is only kernel stuffs, OK
  <details><summary>i686</summary>
  
  OLD:
  
  ```sh
  Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... OFF
  bitcoin-qt (GUI) .................... ON
  bitcoin-gui (GUI, multiprocess) ..... OFF
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... ON
  bitcoin-chainstate (experimental) ... OFF
  libbitcoinkernel (experimental) ..... OFF
  kernel-test (experimental) .......... OFF
Optional features:
  wallet support ...................... ON
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. OFF
  USDT tracing ........................ ON
  QR code (GUI) ....................... ON
  DBus (GUI) .......................... ON
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... ON
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
  ```
  
  NEW:
  
  ```sh
  Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... OFF
  bitcoin-qt (GUI) .................... ON
  bitcoin-gui (GUI, multiprocess) ..... OFF
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... ON
  bitcoin-chainstate (experimental) ... ON
  libbitcoinkernel (experimental) ..... ON
  kernel-test (experimental) .......... ON
Optional features:
  wallet support ...................... ON
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. OFF
  USDT tracing ........................ ON
  QR code (GUI) ....................... ON
  DBus (GUI) .......................... ON
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... ON
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
  ```
  
  </details>
  
   "macOS-cross to arm64"  diff is only bitcoin-chainstate stuffs, OK
  <details><summary>mac arm64</summary>
  
  OLD:
  
  ```sh
  Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... ON
  bitcoin-qt (GUI) .................... ON
  bitcoin-gui (GUI, multiprocess) ..... ON
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... ON
  bitcoin-chainstate (experimental) ... OFF
  libbitcoinkernel (experimental) ..... ON
  kernel-test (experimental) .......... ON
Optional features:
  wallet support ...................... ON
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. ON
  USDT tracing ........................ OFF
  QR code (GUI) ....................... ON
  DBus (GUI) .......................... OFF
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... ON
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
  ```
  
  NEW:
  
  ```sh
  Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... ON
  bitcoin-qt (GUI) .................... ON
  bitcoin-gui (GUI, multiprocess) ..... ON
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... ON
  bitcoin-chainstate (experimental) ... ON
  libbitcoinkernel (experimental) ..... ON
  kernel-test (experimental) .......... ON
Optional features:
  wallet support ...................... ON
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. ON
  USDT tracing ........................ OFF
  QR code (GUI) ....................... ON
  DBus (GUI) .......................... OFF
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... ON
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
  ```
  
  </details>
  
   "macOS-cross to x86_64"  diff is only kernel stuffs,  OK
  
  <details><summary>mac x86_64</summary>
  
  OLD:
  
  ```sh
  Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... ON
  bitcoin-qt (GUI) .................... ON
  bitcoin-gui (GUI, multiprocess) ..... ON
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... ON
  bitcoin-chainstate (experimental) ... OFF
  libbitcoinkernel (experimental) ..... OFF
  kernel-test (experimental) .......... OFF
Optional features:
  wallet support ...................... ON
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. ON
  USDT tracing ........................ OFF
  QR code (GUI) ....................... ON
  DBus (GUI) .......................... OFF
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... ON
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
  ```
  
  NEW

```sh
onfigure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... ON
  bitcoin-qt (GUI) .................... ON
  bitcoin-gui (GUI, multiprocess) ..... ON
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... ON
  bitcoin-chainstate (experimental) ... ON
  libbitcoinkernel (experimental) ..... ON
  kernel-test (experimental) .......... ON
Optional features:
  wallet support ...................... ON
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. ON
  USDT tracing ........................ OFF
  QR code (GUI) ....................... ON
  DBus (GUI) .......................... OFF
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... ON
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
```

</details>

 "Alpine (musl)'"  diff is only kernel stuffs,  OK
  
  <details><summary>Alpine (musl)'</summary>
  
  OLD:
  
```sh
Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... ON
  bitcoin-qt (GUI) .................... ON
  bitcoin-gui (GUI, multiprocess) ..... ON
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... ON
  bitcoin-chainstate (experimental) ... OFF
  libbitcoinkernel (experimental) ..... OFF
  kernel-test (experimental) .......... OFF
Optional features:
  wallet support ...................... ON
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. ON
  USDT tracing ........................ ON
  QR code (GUI) ....................... ON
  DBus (GUI) .......................... ON
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... ON
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
```

NEW:

```sh
Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... ON
  bitcoin-qt (GUI) .................... ON
  bitcoin-gui (GUI, multiprocess) ..... ON
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... ON
  bitcoin-chainstate (experimental) ... ON
  libbitcoinkernel (experimental) ..... ON
  kernel-test (experimental) .......... ON
Optional features:
  wallet support ...................... ON
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. ON
  USDT tracing ........................ ON
  QR code (GUI) ....................... ON
  DBus (GUI) .......................... ON
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... ON
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
```

</details>

 "Msan'"  diff is only kernel stuffs,  OK
  
  <details><summary>Msan'</summary>
  
  OLD:
  
  ```sh
  Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... ON
  bitcoin-qt (GUI) .................... OFF
  bitcoin-gui (GUI, multiprocess) ..... OFF
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... ON
  bitcoin-chainstate (experimental) ... OFF
  libbitcoinkernel (experimental) ..... OFF
  kernel-test (experimental) .......... OFF
Optional features:
  wallet support ...................... ON
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. ON
  USDT tracing ........................ ON
  QR code (GUI) ....................... OFF
  DBus (GUI) .......................... OFF
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... OFF
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
  ```
  
  NEW:
  
  ```sh
  Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... ON
  bitcoin-qt (GUI) .................... OFF
  bitcoin-gui (GUI, multiprocess) ..... OFF
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... ON
  bitcoin-chainstate (experimental) ... ON
  libbitcoinkernel (experimental) ..... ON
  kernel-test (experimental) .......... ON
Optional features:
  wallet support ...................... ON
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. ON
  USDT tracing ........................ ON
  QR code (GUI) ....................... OFF
  DBus (GUI) .......................... OFF
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... OFF
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
  ```
  
  </details>
  
  "No wallet'"  no diff,  OK
  
  <details><summary>No wallet</summary>
  
  OLD:
  
  ```sh
  Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... ON
  bitcoin-qt (GUI) .................... ON
  bitcoin-gui (GUI, multiprocess) ..... ON
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... OFF
  bitcoin-chainstate (experimental) ... ON
  libbitcoinkernel (experimental) ..... ON
  kernel-test (experimental) .......... ON
Optional features:
  wallet support ...................... OFF
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. ON
  USDT tracing ........................ ON
  QR code (GUI) ....................... ON
  DBus (GUI) .......................... ON
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... ON
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
  ```
  
  NEW:
  
  ```sh
  Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... ON
  bitcoin-qt (GUI) .................... ON
  bitcoin-gui (GUI, multiprocess) ..... ON
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... OFF
  bitcoin-chainstate (experimental) ... ON
  libbitcoinkernel (experimental) ..... ON
  kernel-test (experimental) .......... ON
Optional features:
  wallet support ...................... OFF
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. ON
  USDT tracing ........................ ON
  QR code (GUI) ....................... ON
  DBus (GUI) .......................... ON
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... ON
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
  ```
  
  </details>
  

  "TSan'"  diff is only kernel stuffs,  OK
 
  <details><summary>TSan</summary>
  
  OLD:
  
  ```sh
  Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... ON
  bitcoin-qt (GUI) .................... OFF
  bitcoin-gui (GUI, multiprocess) ..... OFF
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... ON
  bitcoin-chainstate (experimental) ... OFF
  libbitcoinkernel (experimental) ..... OFF
  kernel-test (experimental) .......... OFF
Optional features:
  wallet support ...................... ON
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. ON
  USDT tracing ........................ ON
  QR code (GUI) ....................... OFF
  DBus (GUI) .......................... OFF
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... OFF
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
  ```
  
  NEW:
  
  ```sh
  Configure summary
=================
Executables:
  bitcoin ............................. ON
  bitcoind ............................ ON
  bitcoin-node (multiprocess) ......... ON
  bitcoin-qt (GUI) .................... OFF
  bitcoin-gui (GUI, multiprocess) ..... OFF
  bitcoin-cli ......................... ON
  bitcoin-tx .......................... ON
  bitcoin-util ........................ ON
  bitcoin-wallet ...................... ON
  bitcoin-chainstate (experimental) ... ON
  libbitcoinkernel (experimental) ..... ON
  kernel-test (experimental) .......... ON
Optional features:
  wallet support ...................... ON
  external signer ..................... ON
  ZeroMQ .............................. ON
  IPC ................................. ON
  USDT tracing ........................ ON
  QR code (GUI) ....................... OFF
  DBus (GUI) .......................... OFF
Tests:
  test_bitcoin ........................ ON
  test_bitcoin-qt ..................... OFF
  bench_bitcoin ....................... ON
  fuzz binary ......................... ON
  ```
  
  </details>
  
  "Valgrind, fuzz'"  no diff ,  OK?
 
  <details><summary>Valgrind, fuzz</summary>
  
  OLD:
  
  ```sh
  Configure summary
=================
Executables:
  bitcoin ............................. OFF
  bitcoind ............................ OFF
  bitcoin-node (multiprocess) ......... OFF
  bitcoin-qt (GUI) .................... OFF
  bitcoin-gui (GUI, multiprocess) ..... OFF
  bitcoin-cli ......................... OFF
  bitcoin-tx .......................... OFF
  bitcoin-util ........................ OFF
  bitcoin-wallet ...................... OFF
  bitcoin-chainstate (experimental) ... OFF
  libbitcoinkernel (experimental) ..... OFF
  kernel-test (experimental) .......... OFF
Optional features:
  wallet support ...................... ON
  external signer ..................... OFF
  ZeroMQ .............................. OFF
  IPC ................................. OFF
  USDT tracing ........................ OFF
  QR code (GUI) ....................... OFF
  DBus (GUI) .......................... OFF
Tests:
  test_bitcoin ........................ OFF
  test_bitcoin-qt ..................... OFF
  bench_bitcoin ....................... OFF
  fuzz binary ......................... ON
  ```
  
  NEW:
  
  ```sh
  Configure summary
=================
Executables:
  bitcoin ............................. OFF
  bitcoind ............................ OFF
  bitcoin-node (multiprocess) ......... OFF
  bitcoin-qt (GUI) .................... OFF
  bitcoin-gui (GUI, multiprocess) ..... OFF
  bitcoin-cli ......................... OFF
  bitcoin-tx .......................... OFF
  bitcoin-util ........................ OFF
  bitcoin-wallet ...................... OFF
  bitcoin-chainstate (experimental) ... OFF
  libbitcoinkernel (experimental) ..... OFF
  kernel-test (experimental) .......... OFF
Optional features:
  wallet support ...................... ON
  external signer ..................... OFF
  ZeroMQ .............................. OFF
  IPC ................................. OFF
  USDT tracing ........................ OFF
  QR code (GUI) ....................... OFF
  DBus (GUI) .......................... OFF
Tests:
  test_bitcoin ........................ OFF
  test_bitcoin-qt ..................... OFF
  bench_bitcoin ....................... OFF
  fuzz binary ......................... ON
  ```
  
  </details>
  
s390 - no clue see note

hebasto reviewed (approved):
ACK fae83611b8ef358ea7aca7070fd7e82dc06f9755, I have reviewed the code and it looks OK.

maflcko commented on ci/test/00_setup_env_mac_cross.sh:
```diff
@@ -17,4 +17,8 @@ export XCODE_BUILD_ID=15A240d
 export GOAL="deploy"
-export BITCOIN_CONFIG="-DBUILD_GUI=ON -DBUILD_KERNEL_LIB=ON -DREDUCE_EXPORTS=ON"
+export BITCOIN_CONFIG="\
+ --preset=dev-mode \
```
This lead to a failure: https://github.com/maflcko/bitcoin-core-nightly/actions/runs/19550555322/job/55988016546#step:12:25:

```

stdout:
2025-11-20T22:23:58.576632Z TestFramework (INFO): PRNG seed is: 4045980459550867813
2025-11-20T22:23:58.577475Z TestFramework (INFO): Initializing test directory /Users/runner/work/_temp/test_runner_₿_🏃_20251120_220520/tool_bitcoin_chainstate_201
2025-11-20T22:23:59.071213Z TestFramework (INFO): Testing bitcoin-chainstate ['/Users/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/bin/bitcoin-chainstate'] with datadir: /Users/runner/work/_temp/test_runner_₿_🏃_20251120_220520/tool_bitcoin_chainstate_201/node0
2025-11-20T22:23:59.083094Z TestFramework (INFO): STDERR: dyld[26686]: Library not loaded: libbitcoinkernel.dylib
  Referenced from: <4C4C4422-5555-3144-A14F-EA2BABC42000> /Users/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/bin/bitcoin-chainstate
  Reason: tried: 'libbitcoinkernel.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibbitcoinkernel.dylib' (no such file), 'libbitcoinkernel.dylib' (no such file), '/Users/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/libbitcoinkernel.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/libbitcoinkernel.dylib' (no such file), '/Users/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/libbitcoinkernel.dylib' (no such file)
2025-11-20T22:23:59.083212Z TestFramework (ERROR): Unexpected exception
Traceback (most recent call last):
  File "/Users/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/test/functional/test_framework/test_framework.py", line 142, in main
    self.run_test()
    ~~~~~~~~~~~~~^^
  File "/Users/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/test/functional/tool_bitcoin_chainstate.py", line 43, in run_test
    self.add_block(datadir, block_one, "Block has not yet been rejected")
    ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/test/functional/tool_bitcoin_chainstate.py", line 34, in add_block
    raise AssertionError(f"Expected stderr output {expected_stderr} does not partially match stderr:\n{stderr}")
AssertionError: Expected stderr output Block has not yet been rejected does not partially match stderr:
dyld[26686]: Library not loaded: libbitcoinkernel.dylib
  Referenced from: <4C4C4422-5555-3144-A14F-EA2BABC42000> /Users/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/bin/bitcoin-chainstate
  Reason: tried: 'libbitcoinkernel.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibbitcoinkernel.dylib' (no such file), 'libbitcoinkernel.dylib' (no such file), '/Users/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/libbitcoinkernel.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/libbitcoinkernel.dylib' (no such file), '/Users/runner/work/bitcoin-core-nightly/bitcoin-core-nightly/libbitcoinkernel.dylib' (no such file)

janb84 replied:
Weird, dev-mode should have those flags on.

Your build is reporting this [config](https://github.com/maflcko/bitcoin-core-nightly/actions/runs/19550555322/job/55988016371#step:6:1874). 
```sh
  bitcoin-chainstate (experimental) ... ON
  libbitcoinkernel (experimental) ..... ON
  kernel-test (experimental) .......... ON
```

[creating the dylib](https://github.com/maflcko/bitcoin-core-nightly/actions/runs/19550555322/job/55988016371#step:6:2584) [ 59%] Linking CXX shared library ../../lib/libbitcoinkernel.dylib

maflcko replied:
Hmm, this was fixed by setting `DYLD_LIBRARY_PATH`.

Fixed in https://github.com/maflcko/bitcoin-core-nightly/commit/c35885a1904c16918af9602e47a456b15b15e941
